### PR TITLE
Update WalletConnection.ts

### PR DESCRIPTION
**Pull request by wiku-immutable**

### CODE REVIEW

File: src/utils/WalletConnection.ts [hunk 1]
```diff
@@ -45,15 +45,17 @@ export async function generateSpecificWalletConnection(ethSigner: Signer) {
...
-      return true
-    } else {
-      return false
-    }
-  } catch (error) {
-    throw new Error('Error in user registration');
+    await client.getUser(wallet.address);
+    console.log("User is already registered")
+  } catch(e) {
+      console.log(e);
```
Comment (jessemillman) [inline src/utils/WalletConnection.ts]: use console.error()

File: src/utils/WalletConnection.ts [hunk 2]
```diff
@@ -45,15 +45,17 @@ export async function generateSpecificWalletConnection(ethSigner: Signer) {
...
+      try {
```
Comment (jessemillman) [inline src/utils/WalletConnection.ts]: you're catching a generic error, so there could be a case where the user is registered but something else went wrong (i.e. networking).

File: src/utils/WalletConnection.ts [hunk 3]
```diff
@@ -45,15 +45,17 @@ export async function generateSpecificWalletConnection(ethSigner: Signer) {
...
-    if (!isRegistered) {
-      await client.registerOffchain(wallet.walletConnection);
-      return true
-    } else {
-      return false
-    }
-  } catch (error) {
-    throw new Error('Error in user registration');
+    await client.getUser(wallet.address);
+    console.log("User is already registered")
```
Comment (jessemillman) [inline src/utils/WalletConnection.ts]: user should only be considered registered if `user.accounts.length > 0`

**Review by jessemillman — CHANGES_REQUESTED** [review]

File: src/utils/WalletConnection.ts (hunk 3 above)
Comment (wiku-immutable) [reply src/utils/WalletConnection.ts]: I've added this

File: src/utils/WalletConnection.ts (hunk 2 above)
Comment (wiku-immutable) [reply src/utils/WalletConnection.ts]: Can u pls check most recent commit to see if amended version suits?

**Review by samueltan-immutable — APPROVED** [review]

**Pull request merged**